modules/tinymce/src/themes/silver/main/ts/ui/dialog/Tree.ts

```diff
@@ -0,0 +1,177 @@
+const renderIcon = (iconName: string, iconsProvider: Icons.IconProvider, behaviours: Array<Behaviour.NamedConfiguredBehaviour<any, any, any>>): SimpleSpec =>
+  Icons.render(iconName, {
+    tag: 'span',
+    classes: [
+      'tox-tree__icon-wrap',
+      'tox-icon',
+    ],
+    behaviours
+  }, iconsProvider);
+
+const renderIconFromPack = (iconName: string, iconsProvider: Icons.IconProvider): SimpleSpec =>
+  renderIcon(iconName, iconsProvider, []);
+
+const renderDirectoryLabel = (directory: Dialog.Directory, level: number, backstage: UiFactoryBackstage): AlloyButtonSpec => {
+  const internalMenuButton = directory.menu.map((btn) => renderMenuButton(btn, 'tox-mbtn', backstage, Optional.none()));
+  const components: SimpleSpec[] = [
+    {
+      dom: {
+        tag: 'div',
+        classes: [ 'tox-chevron' ]
+      },
+      components: [
+        renderIconFromPack('chevron-right', backstage.shared.providers.icons),
+      ]
+    },
+    {
+      dom: {
+        tag: 'span',
+        styles: {
+          'padding-left': '4px'
+        }
+      },
+      components: [ renderLabel(directory.title) ]
+    }
+  ];
+  internalMenuButton.each((btn) => {
+    components.push(btn);
+  });
+  return AlloyButton.sketch({
+
+    dom: {
+      tag: 'div',
+      classes: [ `tox-tree--directory__label`, 'tox-trbtn' ],
+      styles: {
+        'padding-left': `${level * 8}px`,
+        'font-weight': '700',
+      }
+    },
+    components,
+    action: (button) => {
+      SelectorFind.sibling(button.element, '.tox-tree--directory__children').each((childrenEle) => {
+        button.getSystem().getByDom(childrenEle).each((childrenComp) => Toggling.toggle(childrenComp));
+      });
+    },
+    buttonBehaviours: Behaviour.derive([
+      Toggling.config({
+        toggleOnExecute: true,
+        toggleClass: 'tox-tree--directory__label--active'
+      }),
+      Tabstopping.config({})
+    ])
+  });
+};
+
+const renderDirectoryChildren = (children: Dialog.TreeItem[], level: number, onLeafAction: OnLeafAction, backstage: UiFactoryBackstage): SimpleSpec => {
+  return {
+    dom: {
+      tag: 'div',
+      classes: [ 'tox-tree--directory__children' ],
+      styles: {
+        'padding-left': `${level * 8}px`
+      }
+    },
+    components: children.map((item) => {
+      return item.type === 'leaf' ? renderItemLabel(item, level + 1, onLeafAction, backstage) : renderDirectory(item, level + 1, onLeafAction, backstage);
+    }),
+    behaviours: Behaviour.derive([
+      Toggling.config({
+        toggleClass: 'tox-trbtn--expanded'
+      }),
+      Tabstopping.config({})
+    ])
+  };
+};
+
+const renderDirectory = (dir: Dialog.Directory, level: number, onLeafAction: OnLeafAction, backstage: UiFactoryBackstage): SimpleSpec => {
+  const children =
+    renderDirectoryChildren(dir.children, level, onLeafAction, backstage);
+  return ({
+    dom: {
+      tag: 'div',
+      classes: [ `tox-tree--directory` ],
+      styles: {
+        'display': 'flex',
```

Comment:
These styles should be in the css files.

Reply:
Done!